src/persistent.py

```diff
@@ -0,0 +1,36 @@
+import contextlib
+import os
+from typing import Optional
+
+
+class PersistentString:
+    """A string wrapper which is persistent across reboots.
+
+    Methods:
+        get(value):
+            Gets the value of the string
+        set(value):
+            Sets the value of the string and writes to disk
+    """
+
+    def __init__(self, filepath: str):
+        self.filepath = filepath
+        self._value: Optional[str] = None
+        # Load string if it exists
+        with contextlib.suppress(FileNotFoundError):
+            with open(self.filepath) as f:
+                self._value = f.read()
+
+    def set(self, value: Optional[str]):
+        self._value = value
+
+        # Either save string, or delete state file if None
+        if self._value is None:
```

Comment:
I think this conditional might be backwards?

Reply:
not sure how that happened